gitapp/lib/users_list/users_presenter.dart

```diff
@@ -0,0 +1,23 @@
+import 'package:gitapp/users_list/api_repository.dart';
+import 'package:gitapp/users_list/user_model.dart';
+
+class UsersPresenter {
+  List<User> userList = List.empty(growable: true);
+  List<User> searchedList = List.empty(growable: true);
+  int currentPage = 1;
+  String currentUsername = "";
+
+  UsersPresenter();
+
+  Future<void> updateSearchedListFromApi(String username) async {
```

Comment:
There is some problem or misunderstanding.

Users_presenter should be responsible for the logic. Not only for data searching.

Reply:
Users_presenter now sends state to the view which tells it what to show on screen